### PR TITLE
[FEATURE] Use environment keys for RichTextKeyboardToolbar styles and configuration

### PR DESCRIPTION
**Pull request by r-jarvis**
This PR is in response to comments on this PR after merge: https://github.com/danielsaidi/RichTextKit/pull/140

I split out configuration and style into separate files and implemented the usage of `EnvironmentKeys`/`EnvironmentValues` in order to not require us to inject config/styles in the `RichTextKeyboardToolbar` initializer. 

I did not make the `RichTextKeyboardToolbarStyle` and `RichTextKeyboardToolbarStyleKey` children of `RichTextKeyboardToolbar` because of issues with generics. The `EnvironmentKey` protocol requires concrete types and not generics. Since the generic types aren't actually used in the configuration and style this shouldn't cause any issues. There may be a way around this? Let me know if you'd like me to look into that deeper.

Added a few commented out lines to the iOS demo app in order to show usage. I'm not sure if that is the preferred method or not.

### CODE REVIEW

**Comment by r-jarvis** [comment]
Need to fix tests...

**Comment by DominikBucher12** [comment]
Test fixed here: https://github.com/danielsaidi/RichTextKit/pull/146/files#r1495656071

File: Demo/iOS/EditorScreen.swift [hunk 1]
```diff
@@ -62,6 +62,10 @@ private extension EditorScreen {
             // sheet.colorPickers = .allCases
             return sheet
         }
+        // Uncomment to change keyboard configuration
+        // .richTextKeyboardToolbarConfiguration(.init())
+        // Uncomment to change keyboard style
+        // .richTextKeyboardToolbarStyle(.init())
```
Comment (DominikBucher12) [inline Demo/iOS/EditorScreen.swift:68]: maybe delete this entirely?

**Comment by danielsaidi** [comment]
Hey @r-jarvis - that's great! I will take a look at this on Monday 👍

**Review by danielsaidi — APPROVED** [review]

**Comment by danielsaidi** [comment]
Thank you for the PR @r-jarvis - I've merged it and will modify it a bit before 1.0 🎉

**Pull request merged**